packages/cache/src/cache.ts

```diff
@@ -185,14 +175,32 @@ export async function saveCache(
     const fileSizeLimit = 10 * 1024 * 1024 * 1024 // 10GB per repo limit
     const archiveFileSize = utils.getArchiveFileSizeInBytes(archivePath)
     core.debug(`File Size: ${archiveFileSize}`)
-    if (archiveFileSize > fileSizeLimit) {
+    if (archiveFileSize > fileSizeLimit && !utils.isGhes()) {
       throw new Error(
         `Cache size of ~${Math.round(
           archiveFileSize / (1024 * 1024)
         )} MB (${archiveFileSize} B) is over the 10GB limit, not saving cache.`
       )
     }
 
+    const cacheSize = utils.isGhes() ? archiveFileSize : undefined
```

Comment:
Can we not send size even for non-GHES? There is no harm and server side can decide whether to process it or not (which currently it will not do). But sending every-time just simplifies the logic a bit here.

Reply:
Yes, now we are sending in both ghes and non ghes case